packages/expo-cli/src/commands/init.ts

```diff
@@ -357,24 +363,47 @@ async function installNodeDependenciesAsync(projectRoot: string, packageManager:
   }
 }
 
-export async function initGitRepoAsync(
+async function checkSkipInitializeGitTree(
   root: string,
-  flags: { silent: boolean; commit: boolean } = { silent: false, commit: true }
+  cdPath: string,
+  flags: { silent: boolean } = { silent: false }
 ) {
   // let's see if we're in a git tree
+  let insideGitTree = false;
+
   try {
-    await spawnAsync('git', ['rev-parse', '--is-inside-work-tree'], {
+    const resultPromise = await spawnAsync('git', ['rev-parse', '--is-inside-work-tree'], {
       cwd: root,
     });
-    !flags.silent && Log.log('New project is already inside of a git repo, skipping git init.');
+
+    insideGitTree = resultPromise.stdout.trim() === 'true';
   } catch (e) {
     if (e.errno === 'ENOENT') {
-      !flags.silent && Log.warn('Unable to initialize git repo. `git` not in PATH.');
+      !flags.silent && Log.warn('Unable to check if within existing git repo. `git` not in PATH.');
       return false;
     }
   }
 
-  // not in git tree, so let's init
+  let shouldSkipInitializeGitTree = false;
+
+  if (insideGitTree) {
+    Log.log(
+      `Your project will be initialized inside an existing git repository (${process.cwd()})`
+    );
+    shouldSkipInitializeGitTree =
+      program.nonInteractive ||
```

Comment:
I figured it would be best to default to `shouldSkipInitializeGitTree = true` for non-interactive mode.

My reasoning is that:

1. It seems like most of the comments would like this as the desired behaviour anyway...
2. ...and for those who don't want this behaviour, it's easy enough for them to `git init` themselves afterwards. Not to mention, if they don't realize this behaviour, they'll very quickly realize it the first time they run a git command within the project root.